### ISSUE TITLE
curl API log_client_error results in 504: Gateway time-out

### ISSUE THREAD

**Issue by DrXyzzy** [sagemathinc/cocalc#2184]
Try the following:
```
curl -u sk_xxx: -d error=foo_error https://cocalc.com/api/v1/log_client_error
```

After about 7 seconds, there is an HTML response with title
> cocalc.com | 504: Gateway time-out

I have looked into the problem a bit and notice the following which may be relevant:
1. In src/smc-hub/client.coffee, function `mesg_log_client_error` does not use a callback for error status when it writes to the database, nor does it call e.g. `@success_to_client(id:mesg.id)`.
1. In src/smc-util/message.coffee, the definition of message `log_client_error` does not include the usual optional `id` parameter for message id.

Note that a successful mocha test of a call to `log_client_error` is possible - see #2185 - including read-back of the logged error, but it does not use a callback on the logging step.

**Comment by williamstein** [comment]
This is intentional at the level of the websocket message API.  It’s a waste of bandwidth to send back an acknowledgement message to the user – log_client_error is just fire and forget.  The user doesn’t even know it happened.

That said, naturally the http API, if it also has log_client_error, it should respond with a confirmation that the message was received.

And all that said, I’m fine with changing the websocket api in this case to be a little wasteful and send back an ack, just to keep everything clean and uniform and consistent.

**Comment by DrXyzzy** [comment]
Verified fixed with #2185
